File: engines/foolfuuka/src/chan/content/model/PostsParser.java
package chan.content.model;

import android.net.Uri;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;

import chan.text.ParseException;

public interface PostsParser {
    ArrayList<Posts> convertThreads(InputStream input) throws IOException, ParseException;
    Posts convertPosts(InputStream input, Uri threadUri) throws IOException, ParseException;
    ArrayList<Post> convertSearch(InputStream input) throws IOException, ParseException;
}

File: engines/foolfuuka/src/chan/content/model/PostsParser.kt
package chan.content.model

import android.net.Uri
import chan.text.ParseException
import java.io.IOException
import java.io.InputStream

interface PostsParser {
    @Throws(IOException::class, ParseException::class)
    fun convertThreads(input: InputStream): ArrayList<Posts>

    @Throws(IOException::class, ParseException::class)
    fun convertPosts(
        input: InputStream,
        threadUri: Uri?,
    ): Posts?

    @Throws(IOException::class, ParseException::class)
    fun convertSearch(input: InputStream): ArrayList<Post>
}
